File: dotnet/FinkiRasporedi/Models/Domain/Professor.cs
using FinkiRasporedi.Models.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace FinkiRasporedi.Models.Domain
{
    public class Professor
    {
        [Key]
        public string Id { get; set; }
        public string Name { get; set; }
        [EmailAddress]
        public string? Email { get; set; }
        public ProfessorTitle? ProfessorTitle { get; set; }
        [JsonIgnore]
        public virtual  List<Lecture> Lectures { get; set; }
        [JsonIgnore]
        public int? ScheduleId { get; set; }
        [ForeignKey("ScheduleId")]
        [JsonIgnore]
        public virtual Schedule? Schedule { get; set; }
    }
}

File: flutter/lib/domain/models/professor.dart
import 'enums/professor_title.dart';

class Professor {
  final String id;
  final String name;
  final String? email;
  final ProfessorTitle? professorTitle;

  Professor({
    required this.id,
    required this.name,
    required this.email,
    required this.professorTitle,
  });

  factory Professor.fromJson(Map<String, dynamic> json) {
    return Professor(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      professorTitle: json['professorTitle'] != null
          ? ProfessorTitle.values[json['professorTitle']]
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'professorTitle': professorTitle != null
          ? ProfessorTitle.values.indexOf(professorTitle!)
          : null,
    };
  }
}
